src/paperless_tesseract/parsers.py

```diff
@@ -191,6 +193,29 @@ def _assemble_ocr_sections(self, imgs, middle, text):
         text += self._ocr(imgs[middle + 1:], self.DEFAULT_OCR_LANGUAGE)
         return text
 
+    def get_date(self):
+        text = self.get_text()
+
+        # This regular expression will try to find dates in the document at
+        # hand and will match the following formats:
+        # - XX.YY.ZZZZ with XX + YY being 1 or 2 and ZZZZ being 2 or 4 digits
+        # - XX/YY/ZZZZ with XX + YY being 1 or 2 and ZZZZ being 2 or 4 digits
+        # - XX-YY-ZZZZ with XX + YY being 1 or 2 and ZZZZ being 2 or 4 digits
+        # - XX. MONTH ZZZZ with XX being 1 or 2 and ZZZZ being 2 or 4 digits
+        # - MONTH ZZZZ
+        m = re.search(
+            r'\b([0-9]{1,2})[\.\/-]([0-9]{1,2})[\.\/-]([0-9]{4}|[0-9]{2})\b|' +
+            r'\b([0-9]{1,2}\. [^ ]{3,9} ([0-9]{4}|[0-9]{2}))\b|' +
+            r'\b([^ ]{3,9} [0-9]{4})\b', text)
+
+        if m is None:
+            return None
+
+        return dateparser.parse(m.group(0),
```

Comment:
Ok so I did some digging around on options here re dateutil vs dateparser and I *think* we can do without it if we're a little more creative.  Given that we've already got the date string pre-parsed out for us, we can use dateutil's `parse()` function to do most of the heavy lifting:

```python
from dateutil import parse
return parse(m.group(0), dayfirst=True, yearfirst=True)
```

The `dayfirst=` and `yearfist=` values can be determined by the `DATE_ORDER` setting you setup above.  Something like:

```python
kwargs = {"dayfirst": settings.DATE_ORDER == "MDY", "yearfirst": False}
return parse(m.group(0), **kwargs)
```

Finally, the returned value needs to be made timezone aware, which dateutil won't do for us, but which we can do for ourselves easily enough:

```python
date = parse(m.group(0), **kwargs)
if date.tzinfo is None:
    date = pytz.timezone("UTC").localize(date)
return date
```

What do you think?